src/lib/coerce.js

```diff
@@ -412,6 +412,15 @@ exports.coerce2 = function(containerIn, containerOut, attributes, attribute, dfl
     var propOut = exports.coerce(containerIn, containerOut, attributes, attribute, dflt);
     var valIn = propIn.get();
 
+    var attr = attributes[attribute];
```

Comment:
```
    var attr = nestedProperty(attributes, attribute).get();
```
From `coerce` (line 364 above) - then the `attr || {}` below would be unnecessary - `attr` will always exist.

But I'm a little concerned whether the logic below (`theDefault !== undefined && theDefault !== propOut`) is robust - what if your template explicitly provides the same value as the default? If `valIn` does that, we return `propOut`, but given the logic here if the template does that we'll return `false`.

Could we instead make a base function like `_coerce` that returns both the value and where it came from (default, template, or container), then have `exports.coerce` just ignore that second part and return the value, whereas `exports.coerce2` returns `false` if the value came from a default, otherwise the value?

Actually isn't there another already existing problem with `coerce2`? Looks like if you provide an invalid `valIn`, the prop will be set to the default value but it'll look like you set it explicitly. The base function approach would fix that problem too.

Reply:
@alexcjohnson thanks for the review.
Please find the commits below.